enf-reducer/one-constraint-subsumption.metta

```diff
@@ -0,0 +1,17 @@
+;a function which implements the 1-Constraint-Subsumption Transformation
+(= (oneConstraintSubsume $expOriginal $expRecursive $commandSet) 
+(let* 
+  (
+    ($head (car-atom $expRecursive))
+    ($tail (cdr-atom $expRecursive))
+  )
+  (case $head
+  (
+    (OR $expOriginal)
+    (NOT $expOriginal)
+    (AND (oneConstraintSubsume $expOriginal $tail $commandSet)) ;the POA of this transformation is any non-root AND node
```

Comment:
Doesn't this line of code jump all the `AND` nodes found after the expression, not just the first root node?

Reply:
No, it doesn't I have checked it. Because, I don't think we will have just AND in our expression. It will always be (AND) or (AND A B ...). I have implemented it by first checking if the root node is AND should I change it to that?